trie/stacktrie.go

```diff
@@ -356,9 +376,13 @@ func (t *StackTrie) hash(st *stNode, path []byte) {
 
 	case leafNode:
 		st.key = append(st.key, byte(16))
-		n := shortNode{Key: hexToCompactInPlace(st.key), Val: valueNode(st.val)}
-
-		n.encode(t.h.encbuf)
+		{
```

Comment:
we can also use shortNodeEncoder here?

Reply:
We _could_, but here we know we have a `valueNode`. And the valueNode just does: 
```
func (n valueNode) encode(w rlp.EncoderBuffer) {
	w.WriteBytes(n)
}
```
If we change that to a method which checks the size and theoretically does different things, then the semantics become slightly changed.